workflow/validate/validate.go

```diff
@@ -514,6 +531,51 @@ func validateOutputs(scope map[string]interface{}, tmpl *wfv1.Template) error {
 	return nil
 }
 
+// validateBaseImageOutputs detects if the template contains an output from
+func (ctx *wfValidationCtx) validateBaseImageOutputs(tmpl *wfv1.Template) error {
```

Comment:
The name of this function sounds a little bit unclear for me. Does it validate whether outputs are valid? Maybe, the meaning of `BaseImage` is ambiguous. What do you think?